src/index.js

```diff
@@ -50,7 +51,7 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
     // Avoid dispatching an action if the store is already up-to-date,
     // even if `history` wouldn't do anything if the location is the same
     if(getRouterState().path !== locationToString(location)) {
-      store.dispatch(updatePath(locationToString(location)));
+      store.dispatch(updatePath(locationToString(location), true));
```

Comment:
The idea is that you keep the state up-to-date by listening to the history (lines 49-55), the history update might come from a direct call to history.pushState() or from hitting the back button of the browser. This causes the saved routing state (`state.routing`) to be out of sync with the location. The purpose here is to keep the state updated as soon as we detect a transition. 
The other way around is when we detect a change in state and we need to trigger a transition (lines 58-71), in this case somebody explicitly requested a route change by dispatching the UPDATE_PATH action and we need to react by calling history.pushState().
Please let me know if that makes sense